### PR TITLE
Add a doc describing how to add new RIDs to source-build

### PR DESCRIPTION
**Pull request by omajid**
Anyone else adding their Linux distribution's RID to source-build might find this useful.

I am trying to get this out of my head and into a shape where others can make use of this.

See: #2095

### CODE REVIEW

**Comment by omajid** [comment]
cc @crummel

File: Documentation/add-new-rid.md [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+   -      <Versions>23;24;25;26;27;28;29;30;31;32;33;34</Versions>
+   +      <Versions>23;24;25;26;27;28;29;30;31;32;33;34;35</Versions>
+          <TreatVersionsAsCompatible>false</TreatVersionsAsCompatible>
+        </RuntimeGroup>
+   ```
+
+   Then build with the `/p:UpdateRuntimeFiles=true` flag:
+
+   ```
+   ./build.sh /p:UpdateRuntimeFiles=true
```
Comment (lupin-de-mid) [inline Documentation/add-new-rid.md:114]: Maybe this should be 
 ./build.sh -- /p:UpdateRuntimeFiles=true

Comment (omajid) [reply Documentation/add-new-rid.md:114]: In the dotnet/runtime release/5.0 branch, `./build.sh --help` says:

```
...
Command line arguments starting with '/p:' are passed through to MSBuild.
...
```

Have you run into any cases where `--` was required?

Comment (lupin-de-mid) [reply Documentation/add-new-rid.md:114]: Defenitly yes
```
dotnet-v5.0.104-SDK$ ./build.sh --help
usage: ./build.sh [options]
options:
  --with-ref-packages <dir>          use the specified directory of reference packages
  --with-packages <dir>              use the specified directory of previously-built packages
  --with-sdk <dir>                   use the SDK in the specified directory for bootstrapping
use -- to send the remaining arguments to MSBuild
```

```
dotnet-v5.0.104-SDK$ ./build.sh --with-sdk ../bootstrap/.dotnet /p:UpdateRuntimeFiles=true 
Unrecognized argument '/p:UpdateRuntimeFiles=true'
usage: ./build.sh [options]
options:
  --with-ref-packages <dir>          use the specified directory of reference packages
  --with-packages <dir>              use the specified directory of previously-built packages
  --with-sdk <dir>                   use the SDK in the specified directory for bootstrapping
use -- to send the remaining arguments to MSBuild

```

Comment (omajid) [reply Documentation/add-new-rid.md:114]: Hm. I think we might be talking about different things.

I think you are looking at source-build's `support/tarball/build.sh` (which becomes `./build.sh` when you create the tarball). That needs `--` to separate normal arguments from msbuild arguments.

The step here is for passing the `/p:UpdateRuntimeFiles` flag to dotnet/runtime's build.sh, which doesn't need that: https://github.com/dotnet/runtime/blob/fb2db274ae16120b16c64167cc00538b2acaf155/eng/common/build.sh#L40

Do you know where I could have made that clearer in this doc?

**Review by dseefeld — APPROVED** [review]

File: Documentation/add-new-rid.md (hunk 1 above)
Comment (lupin-de-mid) [reply Documentation/add-new-rid.md:114]: Thank You very match

Comment (lupin-de-mid) [reply Documentation/add-new-rid.md:114]: May be after "1. Grab the source code" add something like "(Not this repository) because I completly miss it :(

Comment (omajid) [reply Documentation/add-new-rid.md:115]: Thanks! Does the new commit make it clearer?

File: Documentation/add-new-rid.md [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+   ```
+   git clone https://github.com/dotnet/runtime
+   cd runtime
+   git checkout release/$version
+   ```
+
+2. Add your new RID to `Microsoft.NETCore.Platforms`'s
+   `runtimeGroups.props` file
+
+   If you are adding a new Linux distro, the chagne might look
```
Comment (crummel) [inline Documentation/add-new-rid.md:76]:
```suggestion
   If you are adding a new Linux distro, the change might look
```

**Review by crummel — APPROVED** [review]
Thanks for writing this up, I think it's be really helpful!

**Pull request merged**